pymongo/server_selectors.py

```diff
@@ -36,54 +43,60 @@ def from_topology_description(cls, topology_description):
             primary,
         )
 
-    def __init__(self, topology_description, server_descriptions, common_wire_version, primary):
+    def __init__(
+        self,
+        topology_description: TopologyDescription,
+        server_descriptions: List[ServerDescription],
+        common_wire_version: Optional[int],
+        primary: Optional[ServerDescription],
+    ):
         self.topology_description = topology_description
         self.server_descriptions = server_descriptions
         self.primary = primary
         self.common_wire_version = common_wire_version
 
-    def with_server_descriptions(self, server_descriptions):
+    def with_server_descriptions(self, server_descriptions: List[ServerDescription]) -> Selection:
         return Selection(
             self.topology_description, server_descriptions, self.common_wire_version, self.primary
         )
 
-    def secondary_with_max_last_write_date(self):
+    def secondary_with_max_last_write_date(self) -> Optional[ServerDescription]:
         secondaries = secondary_server_selector(self)
         if secondaries.server_descriptions:
             return max(secondaries.server_descriptions, key=lambda sd: sd.last_write_date)
         return None
 
     @property
-    def primary_selection(self):
+    def primary_selection(self) -> Selection:
         primaries = [self.primary] if self.primary else []
         return self.with_server_descriptions(primaries)
 
     @property
-    def heartbeat_frequency(self):
+    def heartbeat_frequency(self) -> int:
         return self.topology_description.heartbeat_frequency
 
     @property
-    def topology_type(self):
+    def topology_type(self) -> int:
         return self.topology_description.topology_type
 
-    def __bool__(self):
+    def __bool__(self) -> bool:
         return bool(self.server_descriptions)
 
-    def __getitem__(self, item):
+    def __getitem__(self, item: int) -> ServerDescription:
         return self.server_descriptions[item]
 
 
-def any_server_selector(selection):
+def any_server_selector(selection: Selection) -> Selection:
```

Comment:
What do you think about making this generic since it just returns whatever is provided?